astropy/tests/helper.py

```diff
@@ -470,3 +471,33 @@ def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=None,
     from astropy.units.quantity import _unquantify_allclose_arguments
     np.testing.assert_allclose(*_unquantify_allclose_arguments(
         actual, desired, rtol, atol), **kwargs)
+
+
+def assert_type_is_type(a, b):
```

Comment:
As mentioned above, I find it a bit sad to have a function whose only reason is to avoid style checkers. Maybe we just do not enforce this particular item?

Reply:
I think it is a relatively common mistake to check for types being equal instead of `isinstance`, so maybe leaving this check in place servers some value. If so then it seems we have two choices (1) use `# noqa (E721)` or (2) the new function. 

I'm a little worried about trying to be too clever here with the function as @mhvk pointed out. It's also going to be a bit obscure and it may be difficult to remember to use that. So maybe just using the explicit `noqa` in these cases will convey the intent most clearly?

Being "less clever" with the function as a drop-in replacement would essentially reduce it to `assert a is b`, which does seem a bit silly. With this code even the `# noqa` is not needed since the checker doesn't know `a` and `b` are types.

Reply:
Based on @taldcroft's comment I decided to remove this and just use `# noqa: E721` instead.